builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,104 @@
+
+- Moonbeam [precompiled contracts](builders/pallets-precompiles/precompiles/){target=_blank} such as the Native ERC-20 precompile, XC-20 precompiles, and XCM-related precompiles allow users to manage and transfer assets without requiring access to the EVM. Instead, these actions are done using the Substrate API. So, if your contract holds native tokens or XC-20s and allows arbitrary code execution, these precompiles can be used to drain the balance of the contract
+- Setting a value of `0` when using the `call()` function (for example, `call{value: 0}(...)`) can be overridden by calling the native asset precompile and specifying an amount to transfer in the encoded call data
+- Providing the function selector for the function that you want to allow to be executed is considered safe, assuming that the function call is safe and does not call precompiles
+- Blacklisting contracts that can execute arbitrary call data is not considered safe, as other precompiles might be added in the future. Providing whitelisted contracts that can execute the arbitrary call data is considered safe, assuming that the contracts being called are not precompiles, or that in the case they are, the contract making the call does not hold the native token or any XC-20
+
+In the following sections, you'll learn about each of these security considerations through examples.
+
+### Precompiles Can Override a Set Value {: #setting-a-value }
+
+On Ethereum, a smart contract that allows for arbitrary code execution could force the value of a call to be `0`, guaranteeing that no amount of native currency would be sent with the transaction. Whereas on Moonbeam, setting the value of an arbitrary call to `0` can be overridden by calling the native ERC-20 precompile contract and transferring the native asset through the Substrate API. Since ERC-20s and XC-20s are not native assets, setting a value to `0` doesn't provide any protection for these types of assets on Ethereum or Moonbeam.
+
+For example, if you have a contract that allows arbitrary code execution and you pass it encoded call data that transfers the balance of a contract to another address, you could essentially drain the given contract of it's balance.
+
+To get the encoded call data, you could use a function like the following:
+
+```
+function getBytes(address _contract, address _to) public view returns (bytes memory) {
+    // Load ERC-20 interface of contract
+    IERC20 erc20 = IERC20(_contract);
+    // Get amount to transfer
+    uint256 amount = erc20.balanceOf(address(this));
+    // Build the encoded call data
+    return abi.encodeWithSelector(IERC20.transfer.selector, _to, amount);
+}
+```
```

Comment:
In the first draft, this was within the same contract that makes the arbitrary call. That's not apparent in this case, so the balanceOf call is confusing.

It might be better to just show an attack contract snippet (untested):


```
IERC20 public anyXC20Token;

function attack(address _arbitraryCallContract, address _beneficiary) public {
    // Get amount to transfer out of vulnerable contract
    uint256 amount = anyXC20Token.balanceOf(_arbitraryCallContract);

    // Build attack call data
    bytes attackCallData = abi.encodeWithSelector(IERC20.transfer.selector, _to, amount);

    // Do attack call
    _arbitraryCallContract.makeArbitraryCall(address(anyXC20Token), attackCallData);
}
```

Reply:
Ah yeah I see what you're saying. I'll make this function so it can be in any contract just so it's a bit more flexible